### PR TITLE
[Documentation] Added the On Demand Decision documentation.

### PR DESCRIPTION
**Pull request by vincentpierre**

### CODE REVIEW

File: docs/Learning-Environment-On-Demand-Decision.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# On Demand Decision Making
+
+## Description
+On demand decision making allows agents to request decisions to their
```
Comment (jo3w4rd) [inline docs/Learning-Environment-On-Demand-Decision.md]: request decisions from their brains

File: docs/Learning-Environment-On-Demand-Decision.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+## Description
+On demand decision making allows agents to request decisions to their 
+brains only when needed instead of requesting decisions at a fixed 
+frequency. This is useful when the agents commit to an action for a 
+variable number of steps or when the agents cannot make decisions 
+at the same time.
+
+## How to use
+
+In the agent inspector, there is a new checkbox called
```
Comment (jo3w4rd) [inline docs/Learning-Environment-On-Demand-Decision.md]: delete "new" (it won't be new forever)

File: docs/Learning-Environment-On-Demand-Decision.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+ perform an action every step. In the example above, 
+ `CollectObservations()` will be called every 5 steps and 
+ `AgentAct()` will be called at every step. This means that the 
+ agent will reuse the decision the brain has given it. 
+
+ * If `On Demand Decision` is checked, you are in charge of telling 
+ the agent when to request a decision and when to request an action. 
+ To do so, call the following methods on your agent component.
+   * `RequestDecision()` Call this method to signal the agent that it 
+   must collect his observations and ask the brain for a decision at
```
Comment (jo3w4rd) [inline docs/Learning-Environment-On-Demand-Decision.md]: change "his" to "its"

File: docs/Learning-Environment-On-Demand-Decision.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+ * If `On Demand Decision` is checked, you are in charge of telling 
+ the agent when to request a decision and when to request an action. 
+ To do so, call the following methods on your agent component.
+   * `RequestDecision()` Call this method to signal the agent that it 
+   must collect his observations and ask the brain for a decision at 
+   the next step of the simulation. Note that when an agent requests 
+   a decision, it will also request an action automatically 
+   (This is to ensure that all decisions lead to an action during training)
+   * `RequestAction()` Call this method to signal the agent that 
+   it must reuse his previous action at the next step of the
```
Comment (jo3w4rd) [inline docs/Learning-Environment-On-Demand-Decision.md]: /his/its/s

**Comment by vincentpierre** [comment]
Just addressed the minor fixes needed.

**Pull request merged**